### ISSUE TITLE
kubernetes HPA not justifying the StabilizationWindowSeconds and scaleup policy by pod count

### ISSUE THREAD

**Issue by kandhavelumc** [kubernetes/kubernetes#115586]
### What happened?

I have been facing two weire issues 1.Though the targetCPUUtilizationPercentage breachs the pod not scaling up 2.Pods are not scaled as per the policy

**Though the targetCPUUtilizationPercentage breachs the pod not scaling up ** I have been set the targetCPUUtilizationPercentage as 65 percent and StabilizationWindowSeconds as 60 seconds , Although the targetCPUUtilizationPercentage breached and sustained more than 60 seconds the pods scale up not kicked in as i have checked everytime it scaled above 72 percentage. I have observed the same behaviour in all the application. everytime scale up policy kicked in after 7 to 10 percent breached from the targetCPUUtilizationPercentage.

Pods are not scaled as per the policy

I have been using scaleup policy to scale the pod based on Pods percentage count Below i have given my config

  behavior:
    scaleDown:
      policies:
        - periodSeconds: 15
          type: Pods
          value: 2
      selectPolicy: Min
      stabilizationWindowSeconds: 300
    scaleUp:
      policies:
        - periodSeconds: 15
          type: Pods
          value: 2
        - periodSeconds: 15
          type: Percent
          value: 10
      selectPolicy: Max
      stabilizationWindowSeconds: 60
So when my pod count is below 10 I want to increment the pod by 2 because 10 percent of pod count always 1 , as I need rapid scaling while the pod count is low. But most of the time it increament by 1 for every scaling. Sometimes it scaled by 2 when the currentCPUUtilizationPercentage above 80percent.In the official documentation I have seen the below algorithms for desired replica count desiredReplicas = ceil[currentReplicas * ( currentMetricValue / desiredMetricValue )] I'm trying to override this config through my scaleup policy basically.

Percentage based scaling is working fine I have issue only with pod count based scaling.

I have referred below pages but didn’t get the rationale behind this. https://github.com/kubernetes/enhancements/blob/master/keps/sig-autoscaling/853-configurable-hpa-scale-velocity/README.md#summary https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#scaling-policies

I have attaced my screenshot. Green lines denotes my cpu percentage, red line denoted the target percentage and yellow line denotes horizondal pod count [[enter image description here](https://i.stack.imgur.com/pwOeI.png)](https://i.stack.imgur.com/3qLfE.png)

### What did you expect to happen?

Basically I like to know who there is a lag in scaling the pod and why is not justifying the scaling policy.

### How can we reproduce it (as minimally and precisely as possible)?

Create the HPA and check the pod whether  scaled preciously  against  the given targetpercentage

### Kubernetes version

1.16

### Cloud provider

AWS-EKS

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>
